### PR TITLE
Switch to openni2_launch for cameras.

### PR DESCRIPTION
**Pull request by cburbridge**
This might effect other components if they have not been released since the review. Notably metric maps @RaresAmbrus ?

### CODE REVIEW

File: strands_bringup/launch/strands_cameras.launch [hunk 1]
```diff
@@ -10,23 +10,31 @@
...
-	<include if="$(arg head_camera)" file="$(find openni_wrapper)/launch/main.launch">
-	    <arg name="camera" value="head_xtion" />
-    	<arg name="machine" value="$(arg head_ip)"/>
-		<arg name="user" value="$(arg head_user)" />
+	<include if="$(arg head_camera)" file="$(find strands_bringup)/launch/openni2.launch">
+    	  <arg name="machine" value="$(arg head_ip)"/>
+	  <arg name="user" value="$(arg head_user)" />
+
+	  <arg name="camera" value="head_xtion"  />
+	  <arg name="depth_registration" value="false" />
```
Comment (RaresAmbrus) [inline strands_bringup/launch/strands_cameras.launch]: @cburbridge shouldn't `depth_registration` be set to `true` here?

Comment (cburbridge) [inline strands_bringup/launch/strands_cameras.launch]: yes.

**Comment by marc-hanheide** [comment]
Do we have a list of packages effected? From the review having maybe?

**Comment by cburbridge** [comment]
Charging works fine, as do metric maps... let's merge this?

**Comment by marc-hanheide** [comment]
we should make that merge soon, I suppose. Did you test people perception and V4R stuff? If that works as well I'm happy to merge it.

**Comment by cburbridge** [comment]
It works with V4R, @cdondrup please can you confirm that this does not break people perception?

**Comment by cdondrup** [comment]
I will test it again but it worked well during the review. Just have to change one of the default topic names.

**Comment by marc-hanheide** [comment]
I'm just extra cautious as I need the robot to be working for BBC breakfast on Monday ;-)

**Comment by marc-hanheide** [comment]
@cdondrup ready for merge?

**Comment by cdondrup** [comment]
Let me quickly test this on linda. Slipped my mind I have to admit.

**Comment by cdondrup** [comment]
A few minor things:
- `openni2_launch` has to be a run dependency
- Change depth image default topic in ppl perception to use `/depth/image_rect`
- The `/rgb/image_rect_color` (didn't check the others) has now encoding `rgb8` instead of `bgr8` which for me only results in the boxes around people being blue instead of red, but might hit others as a surprise, e.g. @lucasb-eyer?!

Long story short, for the people perception to work, I just need to change the default depth topic in the launch files once this is merge. So please go ahead after the other issues are resolved.

**Comment by lucasb-eyer** [comment]
Yeah, we started running with openni2 yesterday and no obvious problems so far. I also noticed the change in box-color and was wondering - good to know why!

The head-orientation node will probably become just slightly worse, but I'm close to completely replacing it anyways.

**Comment by cdondrup** [comment]
See https://github.com/strands-project/strands_perception_people/pull/174

**Comment by nilsbore** [comment]
I might have to change some topics in the logging, possibly switching from `openni_wrapper` to `openni2_launch` for re-creating the point clouds from the saved images.

**Comment by cdondrup** [comment]
Any updates on this?

**Comment by bfalacerda** [comment]
can we merge this, I just had some problems because of it not being in

**Comment by marc-hanheide** [comment]
I'm still unclear if all related changes have been done. But maybe now is a good time to enforce it...

**Comment by bfalacerda** [comment]
the new version (still unreleased) of strands_movebase needs openni2_launch. Not sure about all the other camera stuff

**Comment by nilsbore** [comment]
I'm all for merging this and clear up everything needed as we go along. As @bfalacerda said, `strands_movebase` kinda depends on this being merged.

@cburbridge Why is there a need for a separate `openni2.launch` file in this package?

**Comment by cburbridge** [comment]
It is so that we have machine tags compatible with the previous strands_cameras.launch.

**Comment by marc-hanheide** [comment]
So, good to merge?

**Comment by nilsbore** [comment]
We've been running it on Rosie (and it works), though I can't get the machine tags to work for some reason. I have the same problem with other launch files though.

**Comment by marc-hanheide** [comment]
so... I'm about to re-relases scitos_apps, also including this commit: https://github.com/strands-project/scitos_apps/commit/df7aea03f367933eb8171b8930cabce60349bc0a:

```
    ROS_INFO("subscribing to camera topics");
    image_transport::ImageTransport it(*nh);
    subim = it.subscribe("head_xtion/rgb/image_mono", 1, imageCallback);
-   subdepth = it.subscribe("head_xtion/depth/image_rect", 1, depthCallback);
+   subdepth = it.subscribe("head_xtion/depth/image_rect_raw", 1, depthCallback);
    subcamera = nh->subscribe("head_xtion/rgb/camera_info", 1,cameraInfoCallBack);
```

so, it will now also assume the new topic names. This needs to be addresses in the code, I suppose. I won't accep-t the re-release until somebody confirms I should...

**Comment by marc-hanheide** [comment]
This is the PR for re-releasing `scitos_apps` as requested by @cburbridge: https://github.com/strands-project/rosdistro/pull/579

**Comment by marc-hanheide** [comment]
Can @cburbridge, @cdondrup, @lucasb-eyer or @Pandoro, and @nilsbore comment, please? I really lost track what we did change and what not. And if I merge https://github.com/strands-project/rosdistro/pull/579 at least the topic for charging will have changed in the released version. I don't want create an unstable released system. So, please let me know where we stand on this.

**Comment by cdondrup** [comment]
I just tested the charging using the openni2 on Linda. By default it subscribes to:

```
/head_xtion/rgb/image_mono
/head_xtion/depth/image_rect
/head_xtion/rgb/camera_info
```

and works as intended. Hence, the commit should be removed.

**Comment by cburbridge** [comment]
I used this for a while and with the currently released docking code without problem, so agree this review-time commit "Fixing Chaos" should not be released.

**Comment by nilsbore** [comment]
We have also been using openni2_launch with the meta rooms and for chest navigation, works as intended. We have not been running docking though.

**Comment by marc-hanheide** [comment]
ok see https://github.com/strands-project/scitos_apps/pull/151 I will re-release once merged.

**Comment by ferdianjovan** [comment]
Can we merge this now?

**Comment by marc-hanheide** [comment]
Can people please comment on this to finally get round it, asking @gestom @Jailander @lucasb-eyer @Pandoro @nilsbore. Please confirm.

**Comment by lucasb-eyer** [comment]
We've been running with openni2 (not this specific PR) since my last comment, so from Aachen's side this should be fine.

**Comment by nilsbore** [comment]
I'm all for this! Need this for new chest cam navigation.

**Comment by marc-hanheide** [comment]
Right, and I want to things break now rather than later. So here we go.

**Comment by hawesie** [comment]
> On 27 Jan 2016, at 15:45, Marc Hanheide notifications@github.com wrote:
> 
> Right, and I want to things break now rather than later. So here we go.

:)

**Pull request merged**